squid/tests/test_squid.py

```diff
@@ -26,3 +27,33 @@ def test_check_ok(aggregator, check, instance):
     for metric in common.EXPECTED_METRICS:
         aggregator.assert_metric("squid.cachemgr." + metric, tags=expected_tags)
     aggregator.assert_all_metrics_covered()
+
+
+@pytest.mark.parametrize(
+    'raw_version, version_metadata, count',
+    [
+        (
+            'squid/3.0.3',
+            {'version.scheme': 'semver', 'version.major': '3', 'version.minor': '0', 'version.raw': '3.0.3'},
+            5,
+        ),
+        (
+            'squid/1.4.5',
+            {'version.scheme': 'semver', 'version.major': '1', 'version.minor': '4', 'version.raw': '1.4.5'},
+            5,
+        ),
+        ('squid/1.3', {}, 0),
+        ('squid/1', {}, 0,),
+        ('1.4.5', {}, 0,),
```

Comment:
These version can be supported with the parts option: https://github.com/DataDog/integrations-core/pull/5484

Reply:
I added a comment explaining the tests. Let me know if it explains it well enough!